coap/nats/service.go

```diff
@@ -0,0 +1,21 @@
+package nats
+
+import "github.com/mainflux/mainflux"
+
+// Service specifies NATS service API.
+type Service interface {
+	mainflux.MessagePublisher
+	// Subscribes to channel with specified id.
+	Subscribe(string, chan mainflux.RawMessage) (Subscription, error)
+}
+
+// Subscription interface represents subscription to messaging system.
+type Subscription interface {
```

Comment:
You can replace this with function type. Also, can you handle unsubscribe via channels?

Reply:
Worth considering.